provider/kubernetes/kubernetes.go

```diff
@@ -454,18 +458,23 @@ func getTLS(ingress *extensionsv1beta1.Ingress, k8sClient Client) ([]*tls.Config
 	return tlsConfigs, nil
 }
 
-func endpointPortNumber(servicePort corev1.ServicePort, endpointPorts []corev1.EndpointPort) int {
-	if len(endpointPorts) > 0 {
-		// name is optional if there is only one port
-		port := endpointPorts[0]
-		for _, endpointPort := range endpointPorts {
-			if servicePort.Name == endpointPort.Name {
-				port = endpointPort
-			}
+// endpointPortNumber returns the port to be used for this endpoint. It is zero
+// if the endpoint does not match the given service port.
+func endpointPortNumber(servicePort corev1.ServicePort, endpointPorts []corev1.EndpointPort) int32 {
+	// Is this reasonable to assume?
```

Comment:
I'm not sure.

Reply:
Note that I did not modify the logic here; I simply added the comment.

My gut feeling tells me that we shouldn't use the service port if no endpoints are available, but I didn't want to drive-by-change this part along an otherwise unrelated PR.